src/main/java/org/breedinginsight/brapi/v2/services/BrAPIGermplasmService.java

```diff
@@ -153,11 +154,14 @@ public List<Map<String, Object>> processData(List<BrAPIGermplasm> germplasm){
 
     public List<BrAPIGermplasm> getGermplasmByList(UUID programId, String listId) throws ApiException {
         // get list germplasm names
-        BrAPIListDetails listData = brAPIListDAO.getListById(listId, programId).getResult();
-        List<String> germplasmNames = listData.getData();
-
-        // get list BrAPI germplasm variables
-        return germplasmDAO.getGermplasmByRawName(germplasmNames, programId);
+        BrAPIListsSingleResponse listResponse = brAPIListDAO.getListById(listId, programId);
+        if(Objects.nonNull(listResponse)) {
+            BrAPIListDetails listData = listResponse.getResult();
+            List<String> germplasmNames = listData.getData();
```

Comment:
Sorry to be a stickler here, but should probably check that `listData` is not null before calling the `getData()` method

Reply:
Np, you're right of course! I added the other check.